applications/tuxedo_parachain.md

```diff
@@ -0,0 +1,194 @@
+
+### Milestone 2 — Migration of `register_validate_block!` 
+
+- **Estimated Duration:** 2 weeks
+- **FTE:**  1
+- **Costs:** 8,000 USD
+
+The `register_validate_block!` macro is responsible for generating the `validate_block` entrypoint function in the runtime wasm. This is the main entrypoint in the runtime from the relay chain validators' perspective.
+
+We will need to migrate the following to a Tuxedo compatible version
+
+```rust
+cumulus_pallet_parachain_system::register_validate_block! {
+	Runtime = Runtime,
+	BlockExecutor = cumulus_pallet_aura_ext::BlockExecutor::<Runtime, Executive>,
+	CheckInherents = CheckInherents,
+}
+```
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains the difference between the FRAME based approach and the Tuxedo based approach to this macro |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile that can be used to test all the functionality delivered with this milestone. |
+| 1.  | Tuxedo Parachain Template | If you crossed the standalone Tuxedo Template with the Cumulus Template. This is the main deliverable for the grant. A working tempalte node for a Tuxedo-based parachain runtime and colaltor node. |
+
+### Milestone 3 — Benchmarking 
+
+- **Estimated Duration:** 4 weeks
+- **FTE:**  1
+- **Costs:** 16,000 USD
+
+Benchmark common transaction types in standalone and parachain contexts.
+Make good faith comparisons to analogous transaction types in FRAME.
+We hope that the simplicity of the UTXO model will allow performance increases, but right now that is just a theory.
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can spin up the example node and transfer tokens |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile that can be used to test all the functionality delivered with this milestone. |
+| 1.  | Benchmark Report | Full report of throughput for {Transfer, Remark} transactions in {FRAME, Tuxedo} runtimes operated in {Standalone, Parachain} contexts. |
+
+
+## Future Plans
+
+### Near Term
+
+We have a vision for a Tuxedo parachain that acts as an Atomic Swap hub for DOT ecosystem assets to foreign UTXO chains like Monero, Zcash, Cardano, etc.

```

Comment:
How about Bitcoin?

Reply:
Makes sense, thanks for the clarification. 👍 
It might be interesting to compare this approach to Interlay's, though I agree that this shouldn't be the main focus here.